remotes/transport/middleware_kv.go

```diff
@@ -0,0 +1,132 @@
+package transport
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/influxdata/influxdb/v2/kit/platform"
+
+	"github.com/influxdata/influxdb/v2"
+	"github.com/influxdata/influxdb/v2/kv"
+)
+
+var remotesBucket = []byte("remotesv2")
+
+func newTelemetryCollectingService(kv kv.Store, underlying RemoteConnectionService) *telemetryService {
+	return &telemetryService{
+		kv:         kv,
+		underlying: underlying,
+	}
+}
+
+type telemetryService struct {
+	kv         kv.Store
+	underlying RemoteConnectionService
+}
+
+func (t telemetryService) ListRemoteConnections(ctx context.Context, filter influxdb.RemoteConnectionListFilter) (*influxdb.RemoteConnections, error) {
+	return t.underlying.ListRemoteConnections(ctx, filter)
+}
+
+func (t telemetryService) GetRemoteConnection(ctx context.Context, id platform.ID) (*influxdb.RemoteConnection, error) {
+	return t.underlying.GetRemoteConnection(ctx, id)
+}
+
+func (t telemetryService) UpdateRemoteConnection(ctx context.Context, id platform.ID, request influxdb.UpdateRemoteConnectionRequest) (*influxdb.RemoteConnection, error) {
+	return t.underlying.UpdateRemoteConnection(ctx, id, request)
+}
+
+func (t telemetryService) CreateRemoteConnection(ctx context.Context, request influxdb.CreateRemoteConnectionRequest) (*influxdb.RemoteConnection, error) {
+	conn, err := t.underlying.CreateRemoteConnection(ctx, request)
+	if err != nil {
+		return conn, err
+	}
+	if err := t.kv.Update(ctx, func(tx kv.Tx) error {
+		encodedID, err := request.OrgID.Encode()
+		if err != nil {
+			return platform.ErrInvalidID
+		}
+		bucket, err := tx.Bucket(remotesBucket)
+		if err != nil {
+			return err
+		}
+		count, err := t.countRemotes(ctx, request.OrgID)
+		if err != nil {
+			return err
+		}
+		return bucket.Put(encodedID, count)
+	}); err != nil {
+		return nil, fmt.Errorf("updating telemetry failed: %v", err)
+	}
+	return conn, nil
+}
+
+func (t telemetryService) DeleteRemoteConnection(ctx context.Context, id platform.ID) error {
+	rc, err := t.underlying.GetRemoteConnection(ctx, id)
+	if err != nil {
+		return err
+	}
+	orgID := rc.OrgID
+
+	err = t.underlying.DeleteRemoteConnection(ctx, id)
+	if err != nil {
+		return err
+	}
+	if err := t.kv.Update(ctx, func(tx kv.Tx) error {
+		encodedID, err := orgID.Encode()
+		if err != nil {
+			return err
+		}
+		bucket, err := tx.Bucket(remotesBucket)
+		if err != nil {
+			return err
+		}
+		count, err := bucket.Get(encodedID)
+		if err != nil {
+			return err
+		}
+
+		c, err := t.unmarshalCount(count)
+		if err != nil {
+			return err
+		}
+		c--
```

Comment:
Let's just call `countRemotes` here as well, instead of trying to decrement (same for both services)